src/components/Calculator.js

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable */
+import React, { Component } from 'react';
+
+import calculate from './logic/calculate';
+import operate from './logic/operate';
+
+export default class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: 0,
+      next: 0,
+      operation: null,
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({ next: e.target.value });
+  }
+
+  handleClick(e) {
+    if (this.isNumber(e.target.outerText)) {
+      console.log(e.target.outerText);
+      this.setState({ next: e.target.outerText });
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ next: res.next });
+    }
+
+    if (
+      e.target.outerText === '+' ||
+      e.target.outerText === '-' ||
+      e.target.outerText === 'x' ||
+      e.target.outerText === '÷'
+    ) {
+      this.setState({ operation: e.target.outerText });
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ next: res.next });
+      this.setState({ total: res.total });
+    }
+
+    if (e.target.outerText === '=') {
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ ...res });
+      this.setState({ next: res.total });
+    }
+    if (e.target.outerText === '.') {
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      if (Object.entries(res).length !== 0) {
+        this.setState({ ...res });
+        this.setState({ next: res.next });
+      }
+    }
+    if (e.target.outerText === 'AC') {
+      const reset = { total: 0, next: 0, operation: null };
+      this.setState({ ...reset });
+    }
+    if (e.target.outerText === '+/-') {
+      const obj = this.state;
+      const res = calculate(obj, e.target.outerText);
+      this.setState({ ...res });
+      this.setState({ next: res.next });
+    }
+    if (e.target.outerText === '%') {
+      const obj = this.state;
+      this.setState({ next: parseFloat(obj.next / 100).toString() });
+    }
+  }
+
+  isNumber = (item) => !!item.match(/[0-9]+/);
+
+  render() {
+    return (
+      <div className="Calculator">
+        <input
```

Comment:
- [ OPTIONAL ] It would be great if you can think a way to remove these warnings :thought_balloon:
![Screenshot from 2021-10-18 23-40-32](https://user-images.githubusercontent.com/62705947/137811618-876d3586-6983-42ce-8b23-c99483afcceb.png)